client/src/modules/cohort/components/CohortHeader.jsx

```diff
@@ -98,21 +100,22 @@ class CohortHeader extends PureComponent {
         params: { id }
       }
     } = this.props;
-    const { name } = this.state;
-    const nameToUpdate = name.trim();
+    const { nameInputValue } = this.state;
+    const nameToUpdate = nameInputValue.trim();
     const { name: previousName } = details;
 
-    if (previousName === name) {
+    if (previousName === nameToUpdate) {
```

Comment:
check if you doesn't have to trim previousName also.
Down below we are triming previousDescription for some reason